WIP: Update Zoe guides (#70)
Agoric/documentation: main/zoe/guide/contracts/atomic-swap.md

> # Swaps

Reviewer: atomic-swap looks great!

escrow and offer safety. At no time does any user have the ability to
behave opportunistically.

## Atomic Swap

> In the `atomicSwap` contract, anyone who has access to the swap instance can

Reviewer: Ah, I think I forgot to explain this part. When we changed `publicSwap` to `atomicSwap`, we made it invite only, so it's no longer open to the public. 

Let's change this to something like:
```suggestion
In the `atomicSwap` contract, anyone can securely swap with a counterparty by escrowing digital assets with Zoe and sending an invite to the swap to the counterparty.
```

make an offer, with no invites necessary.

> Let's say that Alice wants to create a swap for which anyone can be the

Reviewer: ```suggestion
Let's say that Alice wants to swap with Bob as the counterparty.
```

> } = await zoe.escrow(aliceOfferRules, alicePayments);

Reviewer: Let's update this to `const { seat, payout } = await zoe.redeem(newInvite, aliceOfferRules, alicePayments);`

Reviewer: Ah, ok, I see you have this part below. In that case, we can delete this call to zoe.escrow entirely.

```

In order for Alice to escrow with Zoe she needs to redeem her invite. Once Alice redeems her invite she will receive a `seat` and a promise that resolves to her payout.

```js
const { seat: aliceSeat, payout: alicePayoutP } = await zoe.redeem(
  aliceInvite,
  aliceOfferRules,
  alicePayments,
);
```

Alice then makes the first offer in the swap:

```js
const newInviteP = aliceSeat.makeFirstOffer();
```

> She then spreads the invite widely. Bob hears about Alice's

Reviewer: I think I'm probably contradicting something I wrote in the tests for this contract, but I think we should change this to say that Alice sends the invite directly to Bob.

contract and he to look up the invite to see if it matches Alice's claims.

```js
const inviteAssay = zoe.getInviteAssay();
const bobExclusiveInvite = await inviteAssay.claimAll(newInviteP);
const bobInviteExtent = bobExclusiveInvite.getBalance().extent;

const {
  installationHandle: bobInstallationId,
  terms: bobTerms,
} = zoe.getInstance(bobInviteExtent.instanceHandle);

// Bob does checks

> t.equals(bobInstallationId, installationHandle);

Reviewer: `t.equals` is coming from our test framework, tape, which Bob probably wouldn't have. Let's change these to use `insist` with an expression which should be true.

Author: I've never utilized `insist` so I followed the structure above. Let me know if it needs correcting:

```
insist(bobInstallationId === installationHandle)`wrong installation`;
insist(bobTerms.assays === inviteAssay)`wrong assays`;
insist(bobInviteExtent.offerMadeRules === aliceOfferRules.payoutRules)`wrong payoutRules`;
```

Reviewer: That looks like correct usage of insist to me. Thanks for making this change! I think the second insist statement comparison needs to either be both arrays or both a single assay.

Author: ```
insist(bobTerms.assays[0] === inviteAssay)`wrong assays`;
```
